qemu/tests/block_resize.py

```diff
@@ -181,6 +182,30 @@ def get_md5_of_file(filename):
         session = vm.reboot(session=session)
 
         if os_type == 'linux':
+            # Add following code to solve the disk letter change issue on s390x
+            # virtio-blk, i.e. get new 'mpoint', 'partition' and 'disk'
+            machine_type = params['machine_type']
+            drive_format = params['drive_format']
+            if 's390' in machine_type and drive_format == 'virtio':
+                new_block_size = utils_disk.get_disk_size_linux(session, disk)
+                # Check if the disk letter changes by checking the disk size
+                if new_block_size != block_size:
+                    parts_list = utils_disk.get_parts_list(session)
+                    # Get the index of the disk in the partition list and pop
+                    # it up from the list, also pop up its partitions, i.e.
+                    # system disk partitions; the parts_list format: ['vda',
+                    # 'vda1', 'vdb', 'vdb1', 'vdb2', 'dm-0', 'dm-1']
+                    in_dex = parts_list.index(disk)
+                    parts_list.pop(in_dex)
+                    parts_list.pop(in_dex)
+                    parts_list.pop(in_dex)
```

Comment:
why pop the in_dex three times here, I think one would be enough

Reply:
@fbq815 In fact, the other twice pop are for the partitions of the system disk, i.e. vdb1 and vdb2 in the example. I formerly thought to use another way to remove them from the partition list,  but found this is the easiest way :)